packages/ui/src/serializers/xml/serializers/kaoto-xml-serializer.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { EntityType } from '../../../models/camel/entities';
+import { ElementType, StepXmlSerializer } from './step-xml-serializer';
+import { RestXmlSerializer } from './rest-xml-serializer';
+import { BeansEntity } from '../../../models/visualization/metadata';
+import { BeansXmlSerializer } from './beans-xml-serializer';
+import { RouteDefinition } from '@kaoto/camel-catalog/types';
+import { CamelRouteVisualEntity } from '../../../models';
+import { CamelErrorHandlerVisualEntity } from '../../../models/visualization/flows/camel-error-handler-visual-entity';
+import { CamelRestVisualEntity } from '../../../models/visualization/flows/camel-rest-visual-entity';
+import { CamelInterceptFromVisualEntity } from '../../../models/visualization/flows/camel-intercept-from-visual-entity';
+import { CamelInterceptSendToEndpointVisualEntity } from '../../../models/visualization/flows/camel-intercept-send-to-endpoint-visual-entity';
+import { CamelInterceptVisualEntity } from '../../../models/visualization/flows/camel-intercept-visual-entity';
+import { CamelRouteConfigurationVisualEntity } from '../../../models/visualization/flows/camel-route-configuration-visual-entity';
+import { CamelOnCompletionVisualEntity } from '../../../models/visualization/flows/camel-on-completion-visual-entity';
+import { CamelOnExceptionVisualEntity } from '../../../models/visualization/flows/camel-on-exception-visual-entity';
+import { CamelRestConfigurationVisualEntity } from '../../../models/visualization/flows/camel-rest-configuration-visual-entity';
+
+export type EntityDefinition =
+  | CamelRouteVisualEntity
+  | CamelErrorHandlerVisualEntity
+  | CamelRestVisualEntity
+  | BeansEntity
+  | CamelInterceptFromVisualEntity
+  | CamelInterceptSendToEndpointVisualEntity
+  | CamelInterceptVisualEntity
+  | CamelRouteConfigurationVisualEntity
+  | CamelOnCompletionVisualEntity
+  | CamelOnExceptionVisualEntity
+  | CamelRestConfigurationVisualEntity;
```

Comment:
it might be better to move this type somewhere else, maybe a bit higher in the structure, as this might come handy for other code.